plugins/chat/spec/models/chat/message_spec.rb

```diff
@@ -17,6 +17,98 @@
     subject(:message) { described_class.new(message: "") }
 
     it { is_expected.to validate_length_of(:cooked).is_at_most(20_000) }
+
+    it "validates blocks against a json schema" do
+      message =
+        described_class.new(
+          message: "test",
+          blocks: [{ type: "actions", elements: [{ type: "buttonx" }] }],
+        )
+
+      expect(message).to_not be_valid
+      expect(message.errors.full_messages).to eq(
+        [
+          "Blocks [\"value at `/0/elements/0/type` is not one of: [\\\"button\\\"]\", \"object at `/0/elements/0` is missing required properties: text\"]",
+        ],
+      )
+    end
+
+    it "disallows duplicated action_ids" do
```

Comment:
Same here I think :slightly_smiling_face: It just needs proper contexts to explain what’s happening.